doc/how_tos/custom_checkers.rst

```diff
@@ -37,30 +37,36 @@ Firstly we will need to fill in some required boilerplate:
 
   import astroid
   from astroid import nodes
+  from typing import TYPE_CHECKING
 
   from pylint.checkers import BaseChecker
   from pylint.interfaces import IAstroidChecker
-  from pylint.lint import PyLinter
+
+  if TYPE_CHECKING:
+      from pylint.lint import PyLinter
```

Comment:
We were lazy loading this in the `def register(...` section so let's just put this once here.